addons/l10n_fr_certification/models/account_closure.py

```diff
@@ -0,0 +1,153 @@
+            FROM account_move_line aml
+            JOIN account_journal j ON aml.journal_id = j.id
+            JOIN  (SELECT acc.id FROM account_account acc
+                        JOIN account_account_type t ON t.id = acc.user_type_id
+                        WHERE t.type = 'receivable') AS a
+                  ON a.id = aml.account_id
+            JOIN account_move m ON m.id = aml.move_id
+            WHERE j.type = 'sale'
+                AND aml.company_id = %(company_id)s
+                AND m.state = 'posted'
+                AND m.write_date '''
+
+        if date_start and date_stop:
+            params['date_stop'] = date_stop
+            params['date_start'] = date_start
+            query += 'BETWEEN %(date_start)s AND %(date_stop)s'
+        elif not date_start:
+            params['date_stop'] = date_stop
+            query += '<= %(date_stop)s'
+
+        if avoid_move_ids:
+            params['avoid_move_ids'] = tuple(avoid_move_ids)
+            query += 'AND m.id NOT IN %(avoid_move_ids)s'
+
+        return query, params
+
+    def _query_for_aml(self, company, date_start='', date_stop='', avoid_move_ids=[]):
+        query, params = self._build_query_for_aml(company, date_start, date_stop, avoid_move_ids)
+        self.env.cr.execute(query, params)
+        return self.env.cr.dictfetchall()
+
+    def _compute_amounts(self, frequency, company):
+        interval_dates = self._interval_dates(frequency, company)
+        previous_closure = self.search([
+            ('frequency', '=', frequency),
+            ('company_id', '=', company.id)], limit=1, order='sequence_number desc')
+
+        date_query_start = ''
+        previous_date = False
+        if previous_closure and previous_closure.move_ids:
+            previous_date = previous_closure.move_ids.sorted(key=lambda m: m.write_date)[-1].write_date
+            date_query_start = min(previous_date, interval_dates['interval_from'])
+
+        moves_already_counted = self._query_closures_for_move_ids(frequency, company)
+        aml_fetched = self._query_for_aml(company, date_query_start, interval_dates['date_stop'], moves_already_counted)
+
+        date_interval_start = date_query_start and date_query_start or interval_dates['interval_from']
+        aml_interval = filter(lambda aml: aml['move_write_date'] >= date_interval_start, aml_fetched)
+
+        total_interval = sum(aml['balance'] for aml in aml_interval)
+        total_beginning = sum(aml['balance'] for aml in aml_fetched)
+
+        return {'total_interval': total_interval,
+                'total_beginning': previous_closure.total_beginning + total_beginning,
+                'move_ids': [(4, aml['move_id'], False) for aml in aml_interval],
+                'date_closure_stop': interval_dates['date_stop'],
+                'date_closure_start': previous_date and previous_date or interval_dates['interval_from']}
+
+    def _interval_dates(self, frequency, company):
```

Comment:
1) that method deserves a little docstring
2) that method is wrong I think: if you're running it 
- for a daily frequency but at 4AM: you'll get the last 24h as interval, whereas you probably want the full yesterday day.
- for a monthly frequency, but on the second day of the month (for whatever the reasons), you'll get from the 2nd of previous month to today (2nd of this month), whereas you probably want from last month's first day to the end of the month.

3) why are you using datetime objects? date would be more logical (and not bound to timezones)

Reply:
There was some cases that needed to be covered:
I use write_date because move.date can be in the past relative to now. though the move has to be taken into account in the grand total of any closure, but not in the current period being computed.
Already present moves are dismissed and will not not be duplicated across closures of the same frequency
Since I use write_date, i must have the datetime in python (and then again, frequencies are literal)
Also, the algorithm will actually take any move that happened between the last same frequency closure and now, which could be different from the interval. This is reflected on the date_start field of the closure

Reply:
> I use write_date because move.date can be in the past relative to now. though the move has to be taken into account in the grand total of any closure, but not in the current period being computed.

so what? that's not a argument, you can still do the same with the date...

Reply:
Maybe we should add a posted_date on moves and move_lines ?